pkg/controller/machinepool/machinepool_controller.go

```diff
@@ -309,6 +315,85 @@ func (r *ReconcileRemoteMachineSet) Reconcile(ctx context.Context, request recon
 		return reconcile.Result{}, err
 	}
 
+	// Reconcile local CAPI MachineSets when cd.Spec.MachineManagement.Central enabled
+	if cd.Spec.Platform.AWS != nil && cd.Spec.MachineManagement != nil && cd.Spec.MachineManagement.Central != nil {
+		logger.Info("reconciling local machinesets")
+		return r.reconcileLocalMachineSets(pool, cd, masterMachine, logger)
+	}
```

Comment:
```
if cd.Spec.MachineManagement != nil && cd.Spec.MachineManagement.Central != nil
  switch
    case cd.Spec.Platform.AWS != nil
       // reconcileLocalMachineSets
     default
       // error out
```
i'm thinking just so that we are clearer in the what is the main conditional on spec MachineManagement and then platforms.